### PR TITLE
Rework Account to use delegated types

### PR DESCRIPTION
**Pull request by robzolkos**
Closes https://github.com/maybe-finance/maybe/issues/254

This adds a delegated type for each type of bank account.  It renders a custom new partial depending on the type of bank account selected.

CleanShot 2024-02-02 at 18 19 19@2x

- [x] Custom "new" forms per account type

### CODE REVIEW

File: app/models/account.rb [hunk 1]
```diff
@@ -3,5 +3,7 @@ class Account < ApplicationRecord
 
   belongs_to :family
 
+  delegated_type :accountable, types: %w[ Credit Depository Investment Loan OtherAsset OtherLiability Property Vehicle]
```
Comment (christianrojas) [inline app/models/account.rb:6]:
```suggestion
  delegated_type :accountable, types: %w[Credit Depository Investment Loan OtherAsset OtherLiability Property Vehicle]
```

File: app/models/account.rb [hunk 2]
```diff
@@ -3,5 +3,7 @@ class Account < ApplicationRecord
 
   belongs_to :family
 
+  delegated_type :accountable, types: %w[Credit Depository Investment Loan OtherAsset OtherLiability Property Vehicle]
```
Comment (JoshAntBrown) [inline app/models/account.rb:6]: Looks like we could reduce some duplication here?

```suggestion
  delegated_type :accountable, types: VALID_ACCOUNT_TYPES
```

Comment (robzolkos) [reply app/models/account.rb:6]: Yep, will push shortly.  Just in draft mode as I work through it.

**Comment by robzolkos** [comment]
@Shpigford have a play with this branch if you have a sec.  Let me know if this is the right track.  We can put customized "new" forms per account type if we want too (or keep it simple and create them with name and balance and then peeps can customize stuff in custom edit forms.  or whatever.

File: config/tailwind.config.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ module.exports = {
   content: [
     './public/*.html',
     './app/helpers/**/*.rb',
+    './app/models/**/*.rb',
```
Comment (robzolkos) [inline config/tailwind.config.js:7]: This is temporary - we shall put colors in tailwind config.

**Comment by Shpigford** [comment]
@robzolkos Nice! Each account type will definitely have customized "new" forms.

**Comment by robzolkos** [comment]
@Shpigford ok ready for review.   Btw this nukes the Credit and Depository stuff you added in the last hour or so (sti) in favor of delegated types.

**Comment by Shpigford** [comment]
@robzolkos Can you resolve the schema conflict? Came up because #262 added a new table. Then I'll kick the tires on this.

**Comment by Shpigford** [comment]
Can we drop the Accounts `type` column?

**Comment by robzolkos** [comment]
Yes I think so.  We don't have to do it in this PR but if its no longer needing to be used we can do it now or an upcoming one.

**Comment by robzolkos** [comment]
We also need to get 4 icons for the 4 account types that we didn't have ones for...not sure which icons were being used?

**Comment by Shpigford** [comment]
> Yes I think so. We don't have to do it in this PR but if its no longer needing to be used we can do it now or an upcoming one.

Let's go ahead and drop it just to get it cleared out since we aren't doing STI anymore.

> We also need to get 4 icons for the 4 account types that we didn't have ones for...not sure which icons were being used?

I'll tackle that after this gets merged. We've got the icons in Figma for it.

**Comment by Shpigford** [comment]
@robzolkos Merged. Thanks a ton for tackling this!

**Pull request merged**